File: sdk/python/alcedo_sdk/resources/health.py
from .base import BaseResource


class HealthResource(BaseResource):
    """Plugin core health check resource. Maps to GET /health."""

    async def check(self) -> dict:
        """Get the health status of plugin-core.

        Returns: { "status": str, "core": { "db": str, "docker": str, ... }, "plugins": [...] }
        """
        # HealthResource doesn't use plugin_slug — health is global
        return await self._request("GET", "/health")

File: sdk/alcedo-sdk-rust/src/resources/health.rs
use crate::client::BaseClient;
use crate::error::AlcedoError;
use crate::models::HealthResponse;
use reqwest::Method;

pub struct HealthResource {
    client: BaseClient,
}

impl_new!(HealthResource);

impl HealthResource {
    /// Get the health status of plugin-core.
    pub async fn check(&self) -> Result<HealthResponse, AlcedoError> {
        // Health is global — doesn't use plugin_slug
        let builder = self.client.request(Method::GET, "/health");
        self.client.execute_json(builder).await
    }
}
